apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/connectionurl/parser/URLParser.java

```diff
@@ -29,22 +33,36 @@
  */
 public class URLParser {
 
-    private static final String MYSQL_JDBC_URL_PREFIX = "jdbc:mysql";
-    private static final String ORACLE_JDBC_URL_PREFIX = "jdbc:oracle";
-    private static final String H2_JDBC_URL_PREFIX = "jdbc:h2";
-    private static final String POSTGRESQL_JDBC_URL_PREFIX = "jdbc:postgresql";
+    private static ServiceLoader<ConnectionURLParser> JDBCPARSERS
+        = ServiceLoader.load(ConnectionURLParser.class, URLParser.class
+        .getClassLoader());
```

Comment:
I have concern about using ServiceLoader by `URLParser.class.getClassLoader()`. I think this didn't work in real world. Because the parser implementations, including service define files are in plugin-*.jar, which can't access by any default or webapp classloader.

I think the only way is using ClassLoader from `AgentClassLoader#getDefault` could work.

Reply:
At the same time, I prefer init `JDBCPARSERS` used in `#parser` method at the first time.

Reply:
@ascrutae For safe, please run auto tests ( at least DB, Spring and Tomcat related ) to make sure this pr would break the whole database plugin system.

Reply:
I use a standalone java app with the agent to test and if I use `AgentClassLoader#getDefault` it will result loading a URLparser implementation which is not compatible with URLParser

Reply:
what is your meaning? can you provide more details?